zuul-core/src/main/java/com/netflix/zuul/netty/server/Server.java

```diff
@@ -442,18 +450,20 @@ synchronized private void stop()
         }
     }
 
-    public static final AttributeKey<Map<String, Long>> CONN_TIMING = AttributeKey.newInstance("zuulconntiming");
-    public static final String CONN_ACCEPT = "ACCEPT";
+    /**
+     * Keys should be a short string usable in metrics.
+     */
+    public static final AttributeKey<Attrs> CONN_DIMENSIONS = AttributeKey.newInstance("zuulconndimensions");
 
-    private static final class NewConnHandler extends ChannelInboundHandlerAdapter {
+    private final class NewConnHandler extends ChannelInboundHandlerAdapter {
 
         @Override
         public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
             Long now = System.nanoTime();
-            Map<String, Long> timings = new LinkedHashMap<>();
-            timings.put(CONN_ACCEPT, now);
             final Channel child = (Channel) msg;
-            child.attr(CONN_TIMING).set(timings);
+            child.attr(CONN_DIMENSIONS).set(Attrs.newInstance());
```

Comment:
How are these dimensions going to be set?

Reply:
I was planning on setting them in the handlers as the different stages of negotiation happen.  Things like ipv4/6, http/1 or 2.   Connection names like the target, or authority for outbound ssl.   The channel is available to the handlers, so this should be acccessible.